### PR TITLE
Update constructor in red classes and throw + getSomethingThrowTest

### PR DESCRIPTION
**Pull request by rvadovic**

### CODE REVIEW

File: src/main/java/sk/uniba/fmph/dcs/game_board/ResourceSource.java [hunk 1]
```diff
@@ -14,7 +14,7 @@ public final class ResourceSource implements InterfaceFigureLocationInternal {
     private int maxFigureColours;
     private ArrayList<PlayerOrder> figures;
 
-    public ResourceSource(final String name, final Effect resource) {
+    public ResourceSource(final String name, final Effect resource, CurrentThrow currentThrow, final int playerCount) {
```
Comment (rvadovic) [inline src/main/java/sk/uniba/fmph/dcs/game_board/ResourceSource.java:17]:
```suggestion
    public ResourceSource(final String name, final Effect resource, final CurrentThrow currentThrow, final int playerCount) {
```

File: src/main/java/sk/uniba/fmph/dcs/game_board/ResourceSource.java [hunk 2]
```diff
@@ -14,7 +14,7 @@ public final class ResourceSource implements InterfaceFigureLocationInternal {
     private int maxFigureColours;
     private ArrayList<PlayerOrder> figures;
 
-    public ResourceSource(final String name, final Effect resource) {
+    public ResourceSource(final String name, final Effect resource, final CurrentThrow currentThrow, final int playerCount) {
```
Comment (Fremal31) [inline src/main/java/sk/uniba/fmph/dcs/game_board/ResourceSource.java:17]:
```suggestion
    public ResourceSource(final String name, final Effect resource, final CurrentThrow currentThrow, final int playerCount, final int maxFigureColours) {
```
Toto tu tiez netreba?

File: src/main/java/sk/uniba/fmph/dcs/game_board/CurrentThrow.java [hunk 3]
```diff
@@ -7,7 +7,7 @@ public final class CurrentThrow implements InterfaceToolUse {
     private Effect throwsFor;
     private int throwResult;
 
-    public CurrentThrow() {
+    public CurrentThrow(Throw _throw) {
```
Comment (rvadovic) [inline src/main/java/sk/uniba/fmph/dcs/game_board/CurrentThrow.java:10]:
```suggestion
    public CurrentThrow(final Throw throw1) {
```

File: src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardPlace.java [hunk 4]
```diff
@@ -4,16 +4,21 @@
 import sk.uniba.fmph.dcs.stone_age.Effect;
 import sk.uniba.fmph.dcs.stone_age.HasAction;
 import sk.uniba.fmph.dcs.stone_age.PlayerOrder;
+import sk.uniba.fmph.dcs.stone_age.ImmediateEffect;
+import sk.uniba.fmph.dcs.stone_age.EndOfGameEffect;
+import sk.uniba.fmph.dcs.stone_age.Effect;
+import sk.uniba.fmph.dcs.stone_age.CivilisationCard;
```
Comment (rvadovic) [inline src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardPlace.java:10]:
```suggestion
import sk.uniba.fmph.dcs.stone_age.ImmediateEffect;
import sk.uniba.fmph.dcs.stone_age.EvaluateCivilisationCardImmediateEffect;
import sk.uniba.fmph.dcs.stone_age.CivilisationCard;
```

File: src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardPlace.java [hunk 5]
```diff
@@ -4,16 +4,20 @@
 import sk.uniba.fmph.dcs.stone_age.Effect;
 import sk.uniba.fmph.dcs.stone_age.HasAction;
 import sk.uniba.fmph.dcs.stone_age.PlayerOrder;
+import sk.uniba.fmph.dcs.stone_age.ImmediateEffect;
+import sk.uniba.fmph.dcs.stone_age.EvaluateCivilisationCardImmediateEffect;
```
Comment (rvadovic) [inline src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardPlace.java:8]:
```suggestion
import sk.uniba.fmph.dcs.game_board.EvaluateCivilisationCardImmediateEffect;
```

File: src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardDeck.java [hunk 6]
```diff
@@ -6,7 +6,10 @@
 import java.util.Optional;
 
 public final class CivilizationCardDeck {
-    public CivilizationCardDeck(final Collection<CivilisationCard> cards) {
+    public CivilizationCardDeck(final Collection<CivilisationCard> cards){
```
Comment (rvadovic) [inline src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardDeck.java:9]:
```suggestion
    public CivilizationCardDeck(final Collection<CivilisationCard> cards) {
```

File: src/main/java/sk/uniba/fmph/dcs/game_board/Throw.java [hunk 7]
```diff
@@ -6,6 +6,9 @@ public class Throw {
     private static final int DICESIDES = 6;
     private final int desiredResult; // used for tests, -1 if throw should be random
 
+    public Throw(){
```
Comment (rvadovic) [inline src/main/java/sk/uniba/fmph/dcs/game_board/Throw.java:9]:
```suggestion
    public Throw() {
```

File: src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardPlace.java [hunk 8]
```diff
@@ -4,16 +4,20 @@
 import sk.uniba.fmph.dcs.stone_age.Effect;
 import sk.uniba.fmph.dcs.stone_age.HasAction;
 import sk.uniba.fmph.dcs.stone_age.PlayerOrder;
+import sk.uniba.fmph.dcs.stone_age.ImmediateEffect;
+import sk.uniba.fmph.dcs.game_board.EvaluateCivilisationCardImmediateEffect;
```
Comment (rvadovic) [inline src/main/java/sk/uniba/fmph/dcs/game_board/CivilizationCardPlace.java:8]:
```suggestion
```

File: src/test/java/sk/uniba/fmph/dcs/game_board/GetSomethingThrowTest.java [hunk 9]
```diff
@@ -10,7 +10,8 @@ public class GetSomethingThrowTest {
     @Test
     public void testCalculation() {
         Player p = new Player(null, null);
-        CurrentThrow currentThrow = new CurrentThrow();
+        Throw throw1 = new Throw;
```
Comment (rvadovic) [inline src/test/java/sk/uniba/fmph/dcs/game_board/GetSomethingThrowTest.java:13]:
```suggestion
        Throw throw1 = new Throw();
```

**Review by Fremal31 — APPROVED** [review]
Vyzera dobre 👍

**Pull request merged**